mne_bids/tsv_handler.py

```diff
@@ -44,6 +43,76 @@ def _combine(data1, data2, drop_column=None):
     return data
 
 
+def _combine_cols(data1, data2, col_name, skip_keys=None):
+    """Add two OrderedDict's together and optionally skip certain keys.
+
+    Parameters
+    ----------
+    data1 : collections.OrderedDict
+        Original OrderedDict.
+    data2 : collections.OrderedDict
+        New OrderedDict to be added to the original.
+    col_name : str
+        Name of the column to index over (e.g. `participant_id`
+        in participants.tsv file).
+    skip_keys : list, optional
+        List of column keys to skip writing data on.
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        The new combined data.
+
+    Examples
+    --------
+    Consider `data1` with a table that looks like::
+
+        participant_id | age | sex | test-column
+                    01 | n/a | M | n/a
+                    02 | 23 | n/a | n/a
+
+    and `data2` with a table that looks like::
+
+        participant_id | age | sex | test-column
+                    01 | n/a | n/a | n/a
+                    02 | n/a | n/a | 43
+
+    Then if skip_keys is None, then the output would be::
+
+        participant_id | age | sex | test-column
+                    01 | n/a | n/a | n/a
+                    02 | n/a | n/a | 43
+
+    If skip_keys=`['age', 'sex']`, then the output would be::
+
+        participant_id | age | sex | test-column
+                    01 | n/a | M | n/a
+                    02 | 23 | n/a | 43
```

Comment:
I am sorry for the slowness with this PR.

okay, so now I am thoroughly confused :) Is the purpose only to avoid overwriting entries which are not `n/a` or is it also to preserve user-introduced columns? I want to understand the behavior properly before merging the PR

Reply:
No problem!

Hmm okay maybe this was lost in communication and my addressing of some comments, but I'll try to re-summarize.

Currently in the writing of participants.tsv file, you preserve extra user-introduced columns, but they are by default "n/a" by `_combine_rows` function. Therefore, one would hope to add these columns back, but skip any mne-bids handled columns (e.g. age, sex, hand). 

Therefore `_combine_cols` will add back the original values in the user-introduced columns and allow skipping of certain columns.

The purpose of `skip_keys` is to avoid overwriting entries which are not 'n/a' (i.e. columns  handled by mne-bids).

Reply:
I'm still confused. Let's take a different example than yours. I have:

|participant_id | age | sex |
|--------------|-----|-----|
|                    01 | 27 | n/a | 
|                    02 | 23 | M |

then user adds `test_column`:

|participant_id | age | sex | test-column|
|--------------|-----|-----|-------------|
|                    01 | 27 | n/a | n/a |
|                    02 | 23 | M | 43 |

Now let's say I have another recording with `participant_id = 03`, and I use `mne-bids` to write. The current behavior would be:

|participant_id | age | sex | test-column|
|--------------|-----|-----|-------------|
|                    01 | 27 | n/a | n/a |
|                    02 | 23 | M | n/a |
|                    03 | n/a | F | n/a|

? So the `test-column` is overwritten by `n/a`? Why don't you just add `test-column` to your new data in `mne-bids` and the pass it to `_combine_rows` so you don't overwrite old data with `n/a`? The current solution sounds hackish to me, but maybe I haven't fully understood it yet.

Reply:
Your example is correct in the current behavior, and also your suggestion is also valid.

I guess the soln. that I just pushed is what you said, but still it is essentially the  same code as `_combine_cols`, although I suppose I can remove `_combine_cols` now. I think this way is a bit clearer and doesn't need the generalized `_combine_cols` function.